client/src/components/layouts/default.tsx

```diff
@@ -169,6 +171,7 @@ function DefaultLayout({
   } else {
     return (
       <div className='page-wrapper'>
+        {envData.environment === 'staging' && <StagingWarningModal />}
```

Comment:
```suggestion
        {envData.deploymentEnv === 'staging' &&
          envData.environment === 'production' && <StagingWarningModal />}
```
Both staging and production have the same `environment` value (they're both production environments), so the way to differentiate between them is with `deploymentEnv`.

Also, showing this in development would just be a nuisance, so it should be limited to production.

Reply:
done